### PR TITLE
fix: web api docs to be parsed as markdown rather than mdx FUI-1799

### PR DESCRIPTION
**Pull request by matteematt**
* This work has arose from #1475 which bumped MDX parsing. It now assumes the API docs are MDX files, which they are not, so they contain syntax errors. 
* This PR ammends the settings and copy script to flag that these files are markdown files and to not parse them as MDX
* Then all of the changes required to the api docs are checked in
* I am not going to bump and of the package versions in this PR to make the reviwing easier, this will be part of a subsequent PR
* Updates all normal documentation to `.mdx` extension, and fixes broken links where required
* Doesn't update the previous version docs, but raised this as a discussion point

### CODE REVIEW

**Review by nick-genesis — COMMENTED** [review]
It builds successfully but things are broken I think (imports treated as plain text?)

image

(this branch on the left vs prod on the right)

**Comment by matteematt** [comment]
Nick and I have discussed offline, and it seems the way to fix the issue is just to ensure that the files which are not api docs are turned from `.md` to `.mdx` files. There only remains the question of how we should do that, whether we should just blanket transform them

**Comment by matteematt** [comment]
@nick-genesis I think it should be fixed now, i had to change all file types to mdx (which they should be going forward anyway) and then fixed the links. So now we have clear seperation
1. `.md` files which are the api docs from web, treated as markdown
2. `.mdx` files, which constitute the rest of the files, treated as mdx

Please note I haven't updated the previous version docs. Not sure the effort is worth it as we'll overwrite them all at some point anyway? Interested to hear everyone's thoughts on this

**Review by nick-genesis — APPROVED** [review]
All good now I think

File: plugins/api-docs/processedMap.js [hunk 1]
```diff
@@ -17,5 +17,5 @@ module.exports = {
 	"@genesislcap/foundation-login": "14.137.0",
 	"@genesislcap/foundation-reporting": "14.137.0",
 	"@genesislcap/foundation-testing": "14.137.0",
-	"@genesislcap/foundation-zero": "14.137.0",
+	// "@genesislcap/foundation-zero": "14.137.0",
```
Comment (MrBrunoWolff) [inline plugins/api-docs/processedMap.js:20]: should we just remove this one?

File: plugins/api-docs/src/copyFunctions.ts [hunk 2]
```diff
@@ -36,6 +37,12 @@ export async function createApiDoc(inputFile: string, outputFile: string) {
       })) +
       "\n" +
       content;
+  } else {
```
Comment (MrBrunoWolff) [inline plugins/api-docs/src/copyFunctions.ts:40]: a bit of a weird formatting here but maybe expected?

**Review by MrBrunoWolff — COMMENTED** [review]
did a fresh clone + got a LOT of warnings about broken links, is that expected here?

Local build runs fine (only did build/start scripts, will check prod one)

**Review by MrBrunoWolff — APPROVED** [review]

**Review by wjhendry — APPROVED** [review]

**Pull request merged**